### PR TITLE
HFP-2992 Adjust to meet design created after pull request

### PR DESCRIPTION
**Pull request by otacke**
- Add label translation strings for particular content types
- Add code to override label/description depending on content type

### CODE REVIEW

**Comment by simondate** [comment]
Can you remove this please? https://github.com/otacke/h5p-branching-scenario/blob/HFP-2992-adjust-to-design/semantics.json#L228-L245

File: src/scripts/components/EditorOverlay.js [hunk 1]
```diff
@@ -140,6 +143,69 @@ export default class EditorOverlay extends React.Component {
     H5P.$window.trigger('resize');
   }
 
+  /**
+   * Change field for required finished field to suit current interaction.
+   * @param {object} library Library type object.
+   */
+  modifyrequiresFinishingField(library = {}) {
```
Comment (simondate) [inline src/scripts/components/EditorOverlay.js:150]: modify**R**equiresFinishingField

File: src/scripts/components/EditorOverlay.js [hunk 2]
```diff
@@ -118,6 +118,9 @@ export default class EditorOverlay extends React.Component {
     const titleField = H5PEditor.findField('title', library.metadataForm);
     titleField.$input.on('change', () => this.forceUpdate());
 
+    // Change label and description of "requires finishing" field if they can be more specific.
+    this.modifyrequiresFinishingField(library);
```
Comment (simondate) [inline src/scripts/components/EditorOverlay.js]: modify**R**equiresFinishingField

File: src/scripts/components/EditorOverlay.js (hunk 1 above)
Comment (simondate) [inline src/scripts/components/EditorOverlay.js]: modify**R**equiresFinishingField

**Review by simondate — CHANGES_REQUESTED** [review]
Everything looks good. Just a function that should be changed to camel case.

File: language/en.json [hunk 3]
```diff
@@ -83,6 +83,10 @@
       "configureEndScenario": "Configure the default end scenario screen",
       "jumpToOtherBranch": "Jump to another branch",
       "selectABranchToJump": "Select a branch to jump to",
-      "feedbackDescription": "It is recommended to provide feedback that motivates and also provides guidance. Leave all fields empty if you don't want the user to get feedback after choosing this alternative/viewing this content."
+      "feedbackDescription": "It is recommended to provide feedback that motivates and also provides guidance. Leave all fields empty if you don't want the user to get feedback after choosing this alternative/viewing this content.",
+      "requiresFinishingCoursePresentationLabel": "Required to view entire presentation",
+      "requiresFinishingCoursePresentationDescription": "Urges viewers to navigate all the way to the last slide of the presentation in order to proceed. This setting will not work for active surface mode presentations",
+      "requiresFinishingVideoLabel": "Required to watch until the end",
+      "requiresFinishingVideoDescription": "Urges viewers to complete the video before proceeding with the scenario"
```
Comment (simondate) [inline language/en.json]: Urges -> Forces

File: language/en.json [hunk 4]
```diff
@@ -83,6 +83,10 @@
       "configureEndScenario": "Configure the default end scenario screen",
       "jumpToOtherBranch": "Jump to another branch",
       "selectABranchToJump": "Select a branch to jump to",
-      "feedbackDescription": "It is recommended to provide feedback that motivates and also provides guidance. Leave all fields empty if you don't want the user to get feedback after choosing this alternative/viewing this content."
+      "feedbackDescription": "It is recommended to provide feedback that motivates and also provides guidance. Leave all fields empty if you don't want the user to get feedback after choosing this alternative/viewing this content.",
+      "requiresFinishingCoursePresentationLabel": "Required to view entire presentation",
+      "requiresFinishingCoursePresentationDescription": "Urges viewers to navigate all the way to the last slide of the presentation in order to proceed. This setting will not work for active surface mode presentations",
```
Comment (simondate) [inline language/en.json]: Urges -> Forces

File: src/scripts/components/EditorOverlay.js (hunk 2 above)
Comment (otacke) [reply src/scripts/components/EditorOverlay.js]: Whoops. Well spotted!

**Comment by otacke** [comment]
@simondate I have fixed the camelCase issue and removed the redundant settings field.

**Review by simondate — APPROVED** [review]

**Pull request merged**